### DISCUSSION TITLE
Error creating table widget

### DISCUSSION THREAD

**Discussion by yablochnyi** [filamentphp/filament#14618] [Help]
### Package

Widgets

### Package Version

v3

### How can we help you?

I created a custom page and tried to place a table widget on it, but I get an error. The BalanceOverview widget works fine, but there's an issue with the table widget.
`
class Statistic extends Page
{
    use InteractsWithRecord;

    protected static string $resource = UserResource::class;
    public static ?string $title = 'Статистика';
    protected static ?string $navigationIcon = 'far-chart-line-up';
    protected static string $view = 'filament.resources.user-resource.pages.statistic';

    public function mount(int|string $record): void
    {
        $this->record = $this->resolveRecord($record);
    }

    protected function getHeaderWidgets(): array
    {
        return [
            UserResource\Widgets\BalanceOverview::make(['account' => $this->record->serial_number_user]),
            UserResource\Widgets\Deal::make(['account' => $this->record->serial_number_user])
        ];
    }
}
`

`
class Deal extends BaseWidget
{
    public string $account;
    protected int|string|array $columnSpan = 'full';
    protected static ?string $heading = 'Сделки';

    protected function getTableQuery(): Builder
    {
        return \App\Models\Deal::query()->where('serial_number_user', '=', $this->account);
    }

    protected function getTableColumns(): array
    {
        return [
            Tables\Columns\TextColumn::make('symbol')->label('Актив'),
            Tables\Columns\TextColumn::make('commission')->label('Коммисия'),
            Tables\Columns\TextColumn::make('lots')->label('Объем в крипте'),
            Tables\Columns\TextColumn::make('notional')->label('Объем в деньгах'),
            Tables\Columns\TextColumn::make('price')->label('Цена'),
            Tables\Columns\TextColumn::make('rpnl')->label('PNL'),
            Tables\Columns\TextColumn::make('side')->label('Направление'),
            Tables\Columns\TextColumn::make('timestamp_ms')
                ->formatStateUsing(
                    fn($state) => $state ? \Carbon\Carbon::createFromTimestampMs($state)->toDateTimeString() : null
                )
                ->label('Дата и время'),
        ];
    }

    protected function getTableFilters(): array
    {
        return [
            SelectFilter::make('symbol')
                ->label('Актив')
                ->options(function () {
                    return \App\Models\Deal::query()
                        ->distinct('symbol')
                        ->pluck('symbol', 'symbol')
                        ->all();
                }),
            Filter::make('timestamp_ms')
                ->form([
                    DatePicker::make('from')
                        ->placeholder(fn($state): string => 'Dec 18, ' . now()->subYear()->format('Y')),
                    DatePicker::make('to')
                        ->placeholder(fn($state): string => now()->format('M d, Y')),
                ])
                ->query(function (Builder $query, array $data): Builder {
                    return $query
                        ->when(
                            $data['from'] ?? null,
                            fn(Builder $query, $date): Builder => $query->where(
                                'timestamp_ms',
                                '>=',
                                Carbon::parse($date)->valueOf()
                            ),
                        )
                        ->when(
                            $data['to'] ?? null,
                            fn(Builder $query, $date): Builder => $query->where(
                                'timestamp_ms',
                                '<=',
                                Carbon::parse($date)->endOfDay()->valueOf()
                            ),
                        );
                })
                ->indicateUsing(function (array $data): array {
                    $indicators = [];
                    if ($data['from'] ?? null) {
                        $indicators['from'] = 'From ' . Carbon::parse($data['from'])->toFormattedDateString();
                    }
                    if ($data['to'] ?? null) {
                        $indicators['to'] = 'To ' . Carbon::parse($data['to'])->toFormattedDateString();
                    }

                    return $indicators;
                })
        ];
    }
    
}
`
![Снимок экрана 2024-10-24 в 16 11 24](https://github.com/user-attachments/assets/78612d80-d2db-4e7e-bde8-32ff374839c6)

**Comment by leandrocfe** [comment]
It seems that, for some reason, you don't have a `resources/views/simple-custom.blade.php` file.
Where did you create "simple custom"?
Maybe you could just to create this view...

**Reply by yablochnyi** [reply]
The thing is, I didn't create it and I'm not rendering it anywhere

**Reply by leandrocfe** [reply]
if you are using `filament:cache-components`, try to run `php artisan filament:optimize-clear`
